src/vorta/tray_menu.py

```diff
@@ -73,10 +73,19 @@ def build_menu(self):
         exit_action = menu.addAction(self.tr('Quit'))
         exit_action.triggered.connect(self.app.quit)
 
+    def is_taskbar_dark(self):
+        app = QApplication.instance()
+        system_tray_color = int(QScreen.grabWindow(app.primaryScreen()).toImage().pixel(0, 0))
```

Comment:
This only works if the task bar is on top. So it's true for Mac and Gnome, but not everywhere.

It also triggers a privacy warning on Mac.

![Screenshot 2023-04-18 at 08 24 07](https://user-images.githubusercontent.com/3916435/232702630-2cb43fb5-19a0-4695-b0b0-e2b7ec92f990.jpg)

Reply:
Maybe it's possible to create the task bar icon and then record it's color only?

Reply:
> This only works if the task bar is on top. So it's true for Mac and Gnome, but not everywhere.

I can check which os it's using and check for tray color on either top or bottom based on it.
 
> It also triggers a privacy warning on Mac.

It didn't on mine for some reason.

> Maybe it's possible to create the task bar icon and then record it's color only?

Any ideas on how can I create the initial icon to have a contrasting color to the tray color?

Reply:
> I can check which os it's using and check for tray color on either top or bottom based on it.

KDE Plasma allows the task bar to be on any edge of the screen and it can be shorter than the edge it sits on.

Reply:
> KDE Plasma allows the task bar to be on any edge of the screen and it can be shorter than the edge it sits on.

Oh, Then I think I might have to find another way to locate and scan the tray color.

Reply:
Found a way using `QSystemTrayIcon.geometry()`.

Reply:
I think there is no real way as of now. Getting the following to work looks like the best solution: https://github.com/borgbase/vorta/blob/e0fe766051e521396d7d58f973d1f8002ff749ad/src/vorta/utils.py#L374-L379

Reply:
> I think there is no real way as of now. Getting the following to work looks like the best solution: https://github.com/borgbase/vorta/blob/e0fe766051e521396d7d58f973d1f8002ff749ad/src/vorta/utils.py#L374-L379 

It returns if the system theme is dark or not, but on MacOS the system tray is themed according to the desktop background and is independent of system theme.

Reply:
There is the concept/file type of *template* images on MacOS but I wasn't able to find much information how that could be used from python.

Reply:
> There is the concept/file type of _template_ images on MacOS but I wasn't able to find much information how that could be used from python.

Thanks for the help! I looked into those a little and found a way of using it for our use.

Reply:
> There is the concept/file type of _template_ images on MacOS but I wasn't able to find much information how that could be used from python.

Thanks for the help! I looked into those a little and found a way to use it.
But there's a slight issue, they only work as black and white so the coloured dots which represented the backing up of data in progress don't render in colours and instead just render as white or black. Also I would like to know if this works on linux?

Reply:
I suppose template images do not work in Linux. Let's first add the code for MacOs and think about linux later.

Reply:
> But there's a slight issue, they only work as black and white so the coloured dots which represented the backing up of data in progress don't render in colours and instead just render as white or black.

Can we somehow determine how a template tray icon was rendered and extract information about the tray through that?

Reply:
Alternatively we could add a setting and let the user pick the tray colour manually.

Reply:
> I suppose template images do not work in Linux. Let's first add the code for MacOs and think about linux later.

Are linux system tray's theme also based on the desktop wallpaper or they follow the system's theme? 

> Can we somehow determine how a template tray icon was rendered and extract information about the tray through that?

I tried that but the colour change is based on the system os as the template is a transparent image only having information about the dimensions and the shape and when trying to extract its colour by converting the current qicon to a pixmap and checking the colour of a pixel on there returns same for both light and dark icons.

> Alternatively we could add a setting and let the user pick the tray colour manually.

Yes we can do that as well, but that too will require some logic to check if the selected colour is in range of the dots indicating the working of vorta while backing and converting them to some different colour.

I'll check further into these and let you know if I find any alternatives.

Reply:
> Are linux system tray's theme also based on the desktop wallpaper or they follow the system's theme?

Usually they follow the system theme, but on KDE you can use custom themes that can customize every single bit of the appearance.

> Yes we can do that as well, but that too will require some logic to check if the selected colour is in range of the dots indicating the working of vorta while backing and converting them to some different colour.

This might actually be the best solution.

Reply:
> This might actually be the best solution.

I think so. I'll work on that after I get home in a few days.